### ISSUE TITLE
Minimum cmake version required is (too?) high

### ISSUE THREAD

**Issue by concretevitamin** [google/leveldb#607]
IIUC, a colleague of mine reported that the default cmake version shipped in Ubuntu 16.04 -- arguably a very common version, e.g., popular on EC2 -- is 3.5.1 (he pointed me to https://packages.ubuntu.com/xenial/cmake).  

Is it possible to change leveldb's required cmake version into that?

**Comment by topilski** [comment]
Agree, i changed in my fork to 3.6.3

**Comment by minafarid** [comment]
Seconded. Changing to `cmake_minimum_required(VERSION 3.5.1)` in the main [CMakeLists.txt](https://github.com/google/leveldb/blob/master/CMakeLists.txt#L5) still allows the build and test to work.

**Comment by pwnall** [comment]
Based on our testing matrix, we can't support anything below what's currently in the configuration.

Please feel free to maintain a branch with a different minimum version number. Just know that we won't be able to help if your build breaks.

**Comment by wilhuff** [comment]
Are there any features of CMake 3.9 that are actually required?

CMake 3.9 is barely a year old (https://blog.kitware.com/cmake-3-9-0-available-for-download/) putting it way inside the customer support window we use in customer facing SDKs.

By contrast:
  * Abseil-cpp [requires 3.1](https://github.com/abseil/abseil-cpp/blob/master/CMakeLists.txt#L19)
  * gRPC [requires 2.8](https://github.com/grpc/grpc/blob/master/CMakeLists.txt#L24)
  * GoogleTest [requires 2.6.4](https://github.com/google/googletest/blob/master/CMakeLists.txt#L1)
  * Android NDK r17b requires 3.6
  * OSS Fuzz runs with 3.5.1 

Please reconsider: since Firestore uses CMake this pushes the minimum up well above what we'd otherwise impose on our customers.

**Comment by tmm1** [comment]
Note that one easy way to install a newer cmake is via `pip install cmake`. This worked for me as far back as Ubuntu Trusty 14.04

**Comment by redboarder** [comment]
True. Simple as it is. Trying to compile leveldb under QNX 6.5 that supports the [mostly] painless compilation of CMake 3.6.x.
